elastic/kibana
# [SecuritySolution][Endpoint] Add `scan` response action API

## Summary

Adds a `scan` action response route and related server side logic to handle `scan` action response.

Note: A lot of the changes in the PR are due to test updates that resulted out of adding `scan` command to list of API commands.

### Testing
1. Add `responseActionScanEnabled` feature flag to `xpack.securitySolution.enableExperimental`
2. Run ES/Kibana
3. Run `node x-pack/plugins/security_solution/scripts/endpoint/run_endpoint_agent.js` to start a VM with Elastic Defend installed.
4. Visit `app/security/administration/endpoints` and click on the endpoint on the endpoint list.
5. Copy the endpoint id (`selected_endpopint`) from the URL.
6. Use `curl` to send out a `scan` request to the endpoint with `elastic` user. Use the curl command below:
<details><summary>curl</summary>
<code>
curl --location 'http://localhost:5601/api/endpoint/action/scan' \
--header 'kbn-xsrf: test-xsrf' \
--header 'Content-Type: application/json' \
--header 'Authorization: Basic ZWxhc3RpYzpjaGFuZ2VtZQ==' \
--data '{
    "endpoint_ids": [
        "copied_endpoint_id"
    ],
    "parameters": {
        "path": "/home/ubuntu"
    }
}'
</code>
</details> 

7. You should see the action created on the response actions history for the endpoint.
8. Using any other non existing file path will result in a failed action. UX work to address this will follow.
9. Disabling/removing `responseActionScanEnabled` feature flag should give you a not found error when accessing the API.

### Checklist
